cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+ * See the License for the specific language governin_from_mtxg permissions and
+ * limitations under the License.
+ */
+
+#include "utilities/check_utilities.hpp"
+#include "utilities/conversion_utilities.hpp"
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/graph.hpp>
+#include <cugraph/graph_functions.hpp>
+#include <cugraph/graph_view.hpp>
+#include <cugraph/utilities/high_res_timer.hpp>
+
+#include <raft/core/handle.hpp>
+#include <raft/util/cudart_utils.hpp>
+
+#include <rmm/device_uvector.hpp>
+#include <rmm/mr/device/cuda_memory_resource.hpp>
+
+#include <thrust/set_operations.h>
+
+#include <gtest/gtest.h>
+#include <utilities/base_fixture.hpp>
+#include <utilities/test_graphs.hpp>
+#include <utilities/thrust_wrapper.hpp>
+
+#include <algorithm>
+#include <iterator>
+#include <limits>
+#include <numeric>
+#include <random>
+#include <set>
+#include <vector>
+
+struct KTruss_Usecase {
+  int32_t k{3};
+  bool check_correctness{true};
+  bool test_weighted_{false};
+};
+
+template <typename input_usecase_t>
+class Tests_KTruss : public ::testing::TestWithParam<std::tuple<KTruss_Usecase, input_usecase_t>> {
+ public:
+  Tests_KTruss() {}
+
+  static void SetUpTestCase() {}
+  static void TearDownTestCase() {}
+
+  virtual void SetUp() {}
+  virtual void TearDown() {}
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  std::tuple<rmm::device_uvector<vertex_t>,
+             rmm::device_uvector<vertex_t>,
+             std::optional<rmm::device_uvector<weight_t>>>
+  k_truss_reference(
+    raft::handle_t const& handle,
+    cugraph::graph_view_t<vertex_t, edge_t, false, false> const& graph_view,
+    std::optional<cugraph::edge_property_view_t<edge_t, weight_t const*>> edge_weight_view,
+    edge_t k)
+  {
+    rmm::device_uvector<vertex_t> d_src(0, handle.get_stream());
+    rmm::device_uvector<vertex_t> d_dst(0, handle.get_stream());
+    std::optional<rmm::device_uvector<weight_t>> d_wgt{std::nullopt};
+
+    // Decompress the edgelist - For Rmat generated graph, remove self loop and multi edges
+    std::tie(d_src, d_dst, d_wgt, std::ignore) = decompress_to_edgelist(
+      handle,
+      graph_view,
+      edge_weight_view ? std::make_optional(*edge_weight_view) : std::nullopt,
+      std::optional<cugraph::edge_property_view_t<edge_t, edge_t const*>>{std::nullopt},
+      std::optional<raft::device_span<vertex_t const>>(std::nullopt));
+
+    // get host vectors
+
+    std::vector<vertex_t> h_src(d_src.size());
+    raft::update_host(h_src.data(), d_src.data(), d_src.size(), handle.get_stream());
+    std::vector<vertex_t> h_dst(d_src.size());
+    raft::update_host(h_dst.data(), d_dst.data(), d_dst.size(), handle.get_stream());
+
+    std::vector<weight_t> h_wgt(d_src.size());
+    if (edge_weight_view) {
+      raft::update_host(h_wgt.data(), (*d_wgt).data(), (*d_wgt).size(), handle.get_stream());
+    }
+
+    std::vector<vertex_t> h_src_(h_src.size());
+    std::vector<vertex_t> h_dst_(h_dst.size());
```

Comment:
This code is unnecessary.

Reply:
Right and I am no longer using the coo in the reference implementation.